packages/core-data/src/selectors.ts

```diff
@@ -578,7 +574,10 @@ export const getEntityRecordsTotalPages = (
 	if ( ! queriedState ) {
 		return null;
 	}
-	return getQueriedTotalPages( queriedState, query );
+	if ( query.per_page === -1 ) return 1;
+	const totalItems = getQueriedTotalItems( queriedState, query );
+	if ( ! totalItems ) return totalItems;
+	return Math.ceil( totalItems / query.per_page );
```

Comment:
This can result in `NaN`; the `query.per_page` can be `undefined` when the request relies on the default `per_page` by REST API.

We can try using default `10` as a fallback, but that would be incorrect for some entities that don't support pagination or if the project changes default `per_page` for endpoints.

I was debugging the `useEntityRecords`  hook triggering unstable references warning (`NaN !== NaN`), which led me here.

cc @ntsekouras, @youknowriad

Reply:
Good catch! I guess if the `per_page` is not provided, we can fallback to the header of the response, since it should update properly with other changes of the query. The problem was with the perPage argument not being part of the stable key.  I'll open a PR in a bit.

Reply:
Thank you, @ntsekouras!